### PR TITLE
fix: regression in user secret drain

### PR DESCRIPTION
**Pull request by ycliuhw**
This PR fixes two issues:

- create the `internal` secret backend for a k8s controller because we can have an IaaS model running on a k8s controller.
- the vault policy for user secret drain was `create` and `update`, but it was wrongly changed to `update` and `read` in Dqlite refactoring work, now we revert it back.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

These QA steps were done based on https://github.com/juju/juju/pull/17565

#### k8s model

```
juju model-secret-backend
auto

juju exec --unit snappass-test/0 'secret-add ownedby=snappass-test/0 --owner unit'
secret://f9dd1fcd-0add-4a78-83b1-518b1e4b9a40/cqd1ghjakmoc79v9cvu0

juju exec --unit snappass-test/0 'secret-add ownedby=snappass-test'
secret://f9dd1fcd-0add-4a78-83b1-518b1e4b9a40/cqd1gq3akmoc79v9cvug

juju add-secret foo token=34ae35facd4
secret:cqd1g0bakmoc79v9cvtg

mkubectl -nt1 get secrets -l 'app.kubernetes.io/managed-by = juju'
NAME                               TYPE                                  DATA   AGE
model-exec                         kubernetes.io/service-account-token   3      3m41s
cqd1g0bakmoc79v9cvtg-1             Opaque                                1      2m55s
snappass-test-application-config   Opaque                                5      2m31s
cqd1ghjakmoc79v9cvu0-1             Opaque                                1      2m13s
cqd1gq3akmoc79v9cvug-1             Opaque                                1      99s

juju model-secret-backend myvault1

mkubectl -nt1 get secrets -l 'app.kubernetes.io/managed-by = juju'
NAME                               TYPE                                  DATA   AGE
model-exec                         kubernetes.io/service-account-token   3      5m13s
snappass-test-application-config   Opaque                                5      4m3s

vault kv list -format json t1-${model_uuid: -6}
[
  "cqd1g0bakmoc79v9cvtg-1",
  "cqd1ghjakmoc79v9cvu0-1",
  "cqd1gq3akmoc79v9cvug-1"
]

juju model-secret-backend auto

vault kv list -format json t1-${model_uuid: -6}
{}

mkubectl -nt1 get secrets -l 'app.kubernetes.io/managed-by = juju'
NAME                               TYPE                                  DATA   AGE
model-exec                         kubernetes.io/service-account-token   3      6m22s
snappass-test-application-config   Opaque                                5      5m12s
cqd1g0bakmoc79v9cvtg-1             Opaque                                1      21s
cqd1ghjakmoc79v9cvu0-1             Opaque                                1      21s
cqd1gq3akmoc79v9cvug-1             Opaque                                1      21s
```

## Documentation changes

No

## Links

**Jira card:** [JUJU-5911](https://warthogs.atlassian.net/browse/JUJU-5911)

[JUJU-5911]: https://warthogs.atlassian.net/browse/JUJU-5911?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: domain/secretbackend/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -22,31 +23,41 @@ import (
...
+		return domain.CoerceError(err)
+	}
+}
+
+func createBackend(ctx context.Context, tx *sqlair.TX, backendName string, backendType domainsecretbackend.BackendType) error {
+	backendUUID, err := uuid.NewUUID()
+	if err != nil {
+		return errors.Trace(err)
 	}
+	upsertBackendStmt, err := sqlair.Prepare(`
```
Comment (wallyworld) [inline domain/secretbackend/bootstrap/bootstrap.go:48]: insert

File: domain/secretbackend/bootstrap/bootstrap.go [hunk 2]
```diff
@@ -22,31 +23,41 @@ import (
...
-				ID:                  backendUUID.String(),
-				Name:                backendName,
-				BackendTypeID:       backendType,
-				TokenRotateInterval: internaldatabase.NullDuration{},
-			}).Run()
-			return errors.Annotatef(err, "cannot create secret backend %q", backendName)
-		}))
+			return nil
+		})
+		return domain.CoerceError(err)
```
Comment (wallyworld) [inline domain/secretbackend/bootstrap/bootstrap.go:39]: With the recent PR to automatically coerce txn errors, check if we still need this here

**Review by wallyworld — APPROVED** [review]
Can you please make the jira reference a link

**Comment by wallyworld** [comment]
I assume the 2 failing ci tests on 4.0 are explained by this bug?

**Comment by ycliuhw** [comment]
> I assume the 2 failing ci tests on 4.0 are explained by this bug?

I don't see any secret-related gating test failures. 
The drain tests are currently disabled. They are always green.

**Pull request merged**